src/KickMCMC.jl

```diff
@@ -169,235 +291,160 @@ function createEccentricMCMCModel(observations::Vector{Symbol}, observed_values:
-                end
-                res[:weight][j] += logpdf(dist2,res[:Ω_f][j]+diff)-logpdf(dist1,observed_values[i])
-            elseif obs_symbol == :ω
-                dist1 = WrappedCauchy(res[:ω_f][j], observed_errors[i])
-                dist2 = VonMises(res[:ω_f][j], 1/observed_errors[i]^2)
-                # VonMises works only between μ-π and μ+π, so need to adjust accordingly
-                diff = observed_values[i]-res[:ω_f][j]
-                if diff<-π
-                    diff += 2*π
-                elseif diff>π
-                    diff -= 2*π
-                end
-                res[:weight][j] += logpdf(dist2,res[:ω_f][j]+diff)-logpdf(dist1,observed_values[i])
-            elseif obs_symbol == :i
-                dist1 = Cauchy(res[:i_f][j], observed_errors[i])
-                dist2 = Normal(res[:i_f][j], observed_errors[i])
-                res[:weight][j] += logpdf(dist2,observed_values[i])-logpdf(dist1,observed_values[i])
-            elseif obs_symbol == :v_N
-                dist1 = Cauchy(res[:v_N][j], observed_errors[i])
-                dist2 = Normal(res[:v_N][j], observed_errors[i])
-                res[:weight][j] += logpdf(dist2,observed_values[i])-logpdf(dist1,observed_values[i])
-            elseif obs_symbol == :v_E
-                dist1 = Cauchy(res[:v_E][j], observed_errors[i])
-                dist2 = Normal(res[:v_E][j], observed_errors[i])
-                res[:weight][j] += logpdf(dist2,observed_values[i])-logpdf(dist1,observed_values[i])
-            elseif obs_symbol == :v_r
-                dist1 = Cauchy(res[:v_r][j], observed_errors[i])
-                dist2 = Normal(res[:v_r][j], observed_errors[i])
-                res[:weight][j] += logpdf(dist2,observed_values[i])-logpdf(dist1,observed_values[i])
+    # Add weights to dict
+    logweights = zeros(Float64, nchains, nsamples) # Matrix (nchain x nsample)
+    for i_chain in 1:nchains
+        for i_sample in 1:nsamples
+            for dict_key in keys(loglikelihoods_cauchy)
+                logweights[i_chain, i_sample] = loglikelihoods_normal[dict_key][i_sample, i_chain]
+                                               -loglikelihoods_cauchy[dict_key][i_sample, i_chain]
             end
         end
     end
-    #adjust weights so that maximum weight is unity
-    println(maximum(res[:weight]))
-    res[:weight] .= res[:weight] .- maximum(res[:weight]) #weights are still in log here
-    res[:weight] .= 1 #exp.(res[:weight])
-
-    # if we did a general model, we need to weight the true anomaly
-    if model_type==:general
-        res[:weight] .= res[:weight].*sqrt.(1 .- res[:e_f].^2).^3 ./ (1 .+ res[:e_f].*cos.(res[:ν])).^2
+    logweights = logweights .- maximum(logweights) # set max weights = 1
+    # RTW something broke here
+    weights = exp.(logweights)
+    if (all(isfinite(weights)))
+        results[:weights] = weights 
+    else
+        println("Weights are off!!")
+        results[:weights] = ones(size(weights)) 
```

Comment:
throw exception here?

Reply:
This was written this way to just get an output I could plot, since this catch was happening at the end of a long bit of computation (so if something was broken, I had to rerun the long bit a lot). Once we're happy with the MCMC, I'll replace this with an exception.